comet-java-client/src/test/java/ml/comet/experiment/http/ConnectionUtilsTest.java

```diff
@@ -0,0 +1,123 @@
+
+        this.validateRequest(r, url, params, HttpConstants.Methods.GET, null);
+        assertEquals(HttpConstants.Methods.GET, r.getMethod(), "wrong HTTP method");
+    }
+
+    @Test
+    public void testCreatePostFileRequest() {
+        String url = "http://test.com" + ADD_ASSET;
+        HashMap<String, String> params = new HashMap<String, String>() {{
+            put(EXPERIMENT_KEY, "test_key");
+            put("type", ASSET_TYPE_SOURCE_CODE);
+            put("overwrite", Boolean.toString(false));
+        }};
+        File file = TestUtils.getFile(SOME_TEXT_FILE_NAME);
+        assertNotNull(file, "test file not found");
+
+        Request r = ConnectionUtils.createPostFileRequest(file, url, params);
+        this.validateRequest(r, url, params, HttpConstants.Methods.POST, ConnectionUtils.FORM_MIME_TYPE);
+
+        // check body parts
+        assertEquals(1, r.getBodyParts().size(), "wrong number of body parts");
+        FilePart part = (FilePart) r.getBodyParts().get(0);
+        assertEquals(ConnectionUtils.FILE, part.getName(), "wrong name");
+        assertEquals(ConnectionUtils.FORM_MIME_TYPE, part.getContentType(), "wrong content type");
+        assertEquals(file, part.getFile(), "wrong file");
+    }
+
+    @Test
+    public void testCreatePostByteArrayRequest() {
+        String url = "http://test.com" + ADD_ASSET;
+        HashMap<String, String> params = new HashMap<String, String>() {{
+            put(EXPERIMENT_KEY, "test_key");
+            put("type", ASSET_TYPE_UNKNOWN);
+            put("overwrite", Boolean.toString(true));
+        }};
+        byte[] data = "The test byte data".getBytes();
+
+        Request r = ConnectionUtils.createPostByteArrayRequest(data, url, params);
+        this.validateRequest(r, url, params, HttpConstants.Methods.POST, ConnectionUtils.FORM_MIME_TYPE);
+
+        // check body parts
+        assertEquals(1, r.getBodyParts().size(), "wrong number of body parts");
+        ByteArrayPart part = (ByteArrayPart) r.getBodyParts().get(0);
+        assertEquals(ConnectionUtils.FILE, part.getName(), "wrong name");
+        assertEquals(ConnectionUtils.FORM_MIME_TYPE, part.getContentType(), "wrong content type");
+        assertEquals(data, part.getBytes(), "wrong data array");
+    }
+
+    @Test
+    public void testCreatePostJsonRequest() {
+        String url = "http://test.com" + ADD_ASSET;
+        HtmlRest html = new HtmlRest("<html><body></body></html", "test_key",
+                false, System.currentTimeMillis());
+        String json = JsonUtils.toJson(html);
+
+        Request r = ConnectionUtils.createPostJsonRequest(json, url);
+        this.validateRequest(r, url, null, HttpConstants.Methods.POST, ConnectionUtils.JSON_MIME_TYPE);
+
+        assertEquals(json.length(), r.getBodyGenerator().createBody().getContentLength(), "wrong body");
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {200, 201, 202, 299})
+    public void testIsResponseSuccessfulTrue(int statusCode) {
```

Comment:
if we use a lib for this logic we can remove the test

Reply:
As a result of our sync up meeting, we decided to leave for now.